src/python/gem5/prebuilt/hifiveunmatched/hifive-run.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2022 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+from gem5.resources.resource import Resource, CustomResource
+from gem5.simulate.simulator import Simulator
+from hifive_board import HiFiveUnmatchedBoard
+from typing import List
+
+# collect optional CLI arg for RISCV binary to run
+parser = argparse.ArgumentParser(description="Binary to run on system")
+parser.add_argument(
+    "--riscv_binary", type=str, help="The RISCV binary to execute on the CPU",
+    default="riscv-hello",
+)
+parser.add_argument(
+    "--argv", type=str, help="CLI argument to the binary",
+    default=""
+)
+args = parser.parse_args()
+
+board = HiFiveUnmatchedBoard()
+
+# run default binary if no binary provided, otherwise run CLI provided resource
+if args.riscv_binary == "riscv-hello":
+    board.set_se_binary_workload(
+        Resource(args.riscv_binary)
+    )
+else:
+    board.set_se_binary_workload(
+        CustomResource(args.riscv_binary),
+        arguments=[args.argv]
```

Comment:
You can remove the `if`/`else` if you use `args.argv.split()`

Reply:
The `if`/`else` exists to use Resource() when riscv-hello must be run, and use CustomResource() otherwise. Is there a better way to do that?
Also I pushed a change that uses split() to allow for multiple arguments in case future benchmarks require more than one.

Reply:
I think that it should be a CustomResource in all cases. You the argument is the path to a binary, not the name of a resource.

It would be confusing if you could use the same argument to specify either a resource (of which there is only one? What happens when there's more?) or a path.

Reply:
Ok that makes sense. It is now using CustomResource() only.